Integrate civic 0.4.13

BartDeCaluwe opened the pull request:
This PR will add:

- Civic 0.4.13 integration
- Basic styling of the civic button
- renamed REACT_APP_CLUSTER to NEXT_PUBLIC_REACT_APP_CLUSTER as otherwise nextjs cant read it.

@BouweCeunen I'm assuming that the env var rename has some consequences for our infra?

BartDeCaluwe committed:
feat: add civic 0.4.10

BartDeCaluwe committed:
feat: add styling to identity button

BartDeCaluwe committed:
feat: update civic to v0.4.13

BartDeCaluwe committed:
feat: update devnet and mainnet cd to include env var

BartDeCaluwe committed:
test: fix failing tests

BartDeCaluwe committed:
fix: storybook failing to build

BartDeCaluwe committed:
feat: update stoybook to latest release in order to fix prismjs vuln

BartDeCaluwe committed:
feat: add dist file

RuvendeGroote commented on .github/workflows/cicd.yaml:
```diff
@@ -136,7 +136,7 @@ jobs:
       - name: Build app
         run: |
-          NEXT_PUBLIC_BASE_PATH= yarn build
+          NEXT_PUBLIC_BASE_PATH= NEXT_PUBLIC_REACT_APP_CLUSTER=devnet yarn build
```
remove NEXT_PUBLIC_BASE_PATH

RuvendeGroote commented on .github/workflows/cicd.yaml:
```diff
@@ -199,7 +199,7 @@ jobs:
       - name: Build app
         run: |
-          NEXT_PUBLIC_BASE_PATH= yarn build
+          NEXT_PUBLIC_BASE_PATH= NEXT_PUBLIC_REACT_APP_CLUSTER=mainnet yarn build
```
same

RuvendeGroote commented:
investigate the many rerenders and  duplicate rpc calls

BartDeCaluwe committed:
feat: remove unused env var

BartDeCaluwe committed:
feat: only get the market once

Connecting your wallet causes the client object to change which causes the market to be refetched. This results in unnecessary RPC calls. By having an empty dep array in the useEffect hook, we can ensure the market (and its statistics) is only loaded once.

RuvendeGroote commented:
add info about targetting the right cluster to the readme.
add info about that we use the civic pass (only on devnet) to readme

RuvenSalamon commented on src/components/Identity.tsx:
```diff
@@ -0,0 +1,41 @@
+import { useCredixClient } from "@credix/credix-client";
+import { defaultMarketplace } from "../consts";
+
+export const Identity = () => {
```
Rename to civicprovider (same principle as you did with the clientprovider)

RuvenSalamon replied:
rename to CivicButton

RuvendeGroote commented on src/pages/index.tsx:
```diff
@@ -17,7 +17,7 @@ const Overview: NextPage = () => {
 	useEffect(() => {
 		getMarket();
-	}, [getMarket]);
+	}, []);
```
add a comment explaining why we ignore the eslint warning here and why getmarket should not be added as a dependency again

BartDeCaluwe committed:
feat: remove badge in favour of always showing the civic button + rename identity related components

RuvendeGroote commented on src/config.ts:
```diff
@@ -12,23 +13,26 @@ const localnetConfig: ClusterConfig = {
 	...baseConfig,
 	name: SolanaCluster.LOCALNET,
 	RPCEndpoint: RPCEndpoint.LOCALNET,
+	stage: CivicStage.LOCALNET,
```
this doesn't need to be part of the config. it's very specific for a particular component and can easily be derived from the cluster in the config already.

BartDeCaluwe committed:
feat: add .env files to gitignore

BartDeCaluwe committed:
docs: add cluster selection information

BartDeCaluwe committed:
docs: add section on Identity Verification

BartDeCaluwe committed:
feat: use anchorWallet when creating a client

BartDeCaluwe committed:
feat: add zustand state management

BartDeCaluwe committed:
feat: get market from state

BartDeCaluwe committed:
feat: derive civic stage from cluster

RuvendeGroote commented on src/components/IdentityButton.tsx:
```diff
@@ -0,0 +1,59 @@
+	const [gatekeeperNetwork, setGatekeeperNetwork] = useState<PublicKey>();
+
+	const getGatekeeperNetwork = useCallback(async () => {
+		const market = await client.fetchMarket(defaultMarketplace);
```
can we use the market from the state here?

BartDeCaluwe committed:
feat: get market from state

BartDeCaluwe committed:
test: fix store import path

BartDeCaluwe committed:
feat: add white background color to civic and wallet button
